presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -236,7 +270,15 @@ public static long parseTimestampWithoutTimeZone(String value)
     @Deprecated
     public static long parseTimestampWithoutTimeZone(TimeZoneKey timeZoneKey, String value)
     {
-        return TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
+        try {
+            return java.time.LocalDateTime.parse(value, TIMESTAMP_OPTIONAL_TIMEZONE_FORMATTER).atZone(ZoneId.of(timeZoneKey.getId())).toInstant().toEpochMilli();
+        }
+        catch (DateTimeParseException e) {
+            throw new RuntimeException(e);
```

Comment:
no need to catch; DateTimeParseException is a runtime exception that's more semantic here